packages/grafana-e2e-selectors/src/resolver.ts

```diff
@@ -0,0 +1,84 @@
+import { gte, compare, valid } from 'semver';
+
+import {
+  FunctionSelector,
+  Selectors,
+  SelectorsOf,
+  StringSelector,
+  VersionedSelectorGroup,
+  VersionedSelectors,
+  CssSelector,
+  UrlSelector,
+  FunctionSelector2,
+} from './types';
+
+/**
+ * Resolves selectors based on the Grafana version
+ */
+export function resolveSelectors<T extends Selectors>(
+  versionedSelectors: VersionedSelectorGroup,
+  grafanaVersion = 'latest'
+): SelectorsOf<T> {
+  const version = grafanaVersion.replace(/\-.*/, '');
+
+  return resolveSelectorGroup(versionedSelectors, version);
+}
+
+function resolveSelectorGroup<T extends Selectors>(
+  group: VersionedSelectorGroup,
+  grafanaVersion: string
+): SelectorsOf<T> {
+  const result: Selectors = {};
+
+  for (const [key, value] of Object.entries(group)) {
+    if (isVersionedSelectorGroup(value)) {
+      result[key] = resolveSelectorGroup(value, grafanaVersion);
+    } else {
+      assertIsSemverValid(value, key);
+      result[key] = resolveSelector(value, grafanaVersion);
+    }
+  }
+
+  return result as SelectorsOf<T>;
```

Comment:
i can't think of a way to get rid of this - @joshhunt any ideas? 🤔

Reply:
I had a stab at this, with mapping over Object.entries and passing that into Object.toEntries, but I didn't understand everything enough to get it all working.

Given this isn't runtime code im not _too_ fussed about this.